src/Npgsql/NpgsqlActivitySource.cs

```diff
@@ -20,19 +21,56 @@ static NpgsqlActivitySource()
 
     internal static bool IsEnabled => Source.HasListeners();
 
-    internal static Activity? CommandStart(NpgsqlConnector connector, string sql)
+    internal static Activity? CommandStart(NpgsqlConnector connector, string commandText, CommandType commandType, bool storedProcCompatMode)
```

Comment:
Probably better to just access NpgsqlCommand.EnableStoredProcedureCompatMode directly, rather than passing it as a parameter, this way the code gets optimized by the JIT (it's a pseudo-constant).

Reply:
👍🏻